worker/base/worker.go

```diff
@@ -269,3 +276,34 @@ func (w *Worker) Exporter(name string) (exporter.Exporter, error) {
 func (w *Worker) InstructionCache() instructioncache.InstructionCache {
 	return w.cache
 }
+
+// utility function. could be moved to the constructor logic?
+func Labels(executor, snapshotter string) map[string]string {
+	hostname, err := os.Hostname()
+	if err != nil {
+		hostname = "unknown"
+	}
+	labels := map[string]string{
+		worker.LabelOS:          runtime.GOOS,
+		worker.LabelArch:        runtime.GOARCH,
+		worker.LabelExecutor:    executor,
+		worker.LabelSnapshotter: snapshotter,
+		worker.LabelHostname:    hostname,
+	}
+	return labels
+}
+
+// ID reads the worker id from the metadata store.
+// If not exist, generate a random one,
+func ID(meta *metadata.Store) (string, error) {
```

Comment:
Is it ok to use metadata store here?

Reply:
No, it should only be for snapshot metadata. There is an `All()` request in the snapshots initializer that reads in all records from here and expects them to be snapshots.